demo/api/schema.gql

```diff
@@ -926,6 +926,8 @@ type Mutation {
   updateDamFolder(id: ID!, input: UpdateDamFolderInput!): DamFolder!
   moveDamFolders(folderIds: [ID!]!, targetFolderId: ID, scope: DamScopeInput!): [DamFolder!]!
   deleteDamFolder(id: ID!): Boolean!
+  generateImageTitle(fileId: String!): String!
+  generateAltText(fileId: String!): String!
```

Comment:
those only return the generated text, right? wouldn't a query be more appropriate? (could be even a FieldResolver of DamFile)

Reply:
I guess it's debatable here. My view on it is this:

Pro Query:

- No data is changed on the server

Pro Mutation:

- It doesn't just deliver data from the server but calls a third-party API -> has a "side-effect"
- A new result is generated on every call 
    - -> it should only be called lazy
    - -> it shouldn't be refreshed
    - -> it shouldn't be cached

You can do all that with a query. But from the way you're supposed to use it it feels more like a mutation to me so I'm in favor of mutation here

Reply:
> It doesn't just deliver data from the server but calls a third-party API -> has a "side-effect"

Doing third-party api calls doesn't mean anything. And no, it doesn't have a side effect.

It's just a wording thing, if you have thought about it carefully it's Ok :)